### PR TITLE
Upgrade to angular 16

### PR DESCRIPTION
**Pull request by mshima**
- drop ngx-webstorage dependency.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mraible** [comment]
```
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/common@"^15.0.0" from ngx-webstorage@11.1.1
```

It looks like there's already an issue for this. Not sure if it'll be fixed soon.

https://github.com/PillowPillow/ng2-webstorage/issues/183

**Comment by mshima** [comment]
Yes, working on removing the dependency

File: generators/angular/templates/package.json.ejs [hunk 1]
```diff
@@ -125,7 +124,7 @@
     "webpack-notifier": "<%= nodeDependencies['webpack-notifier'] %>"
   },
   "overrides": {
-    "webpack": "<%= nodeDependencies['webpack'] %>"
+    "webpack": "$webpack"
```
Comment (mshima) [inline generators/angular/templates/package.json.ejs:127]:
```suggestion
    "webpack": "<%= nodeDependencies['webpack'] %>"
```

**Comment by mshima** [comment]
There are a few more changes planned, I will create another PR once ready.

**Comment by mshima** [comment]
`ngx-webstorage` was released, but it was already dropped.
It already cased problems updating to others major angular bumps.

**Comment by mraible** [comment]
> There are a few more changes planned, I will create another PR once ready.

Are you saying there are more changes necessary for Angular 16 or for a v8 beta release?

**Review by mraible — APPROVED** [review]

**Comment by mshima** [comment]
@DanielFran bounty claimed https://opencollective.com/generator-jhipster/expenses/148968

**Comment by DanielFran** [comment]
@mshima approved

**Pull request merged**